src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> headNode;
+    private Node<T> tailNode;
+
+    private Node<T> getNodeFromIndex(int index) {
+        checkIndex(index);
+        Node<T> searchNode = headNode;
+
+        if (index < (size >> 1)) {
+            for (int i = 0; i < index; i++) {
+                searchNode = searchNode.next;
+            }
+        } else {
+            searchNode = tailNode;
+            for (int i = size() - 1; i > index; i--) {
+                searchNode = searchNode.previous;
+            }
+        }
+        return searchNode;
+
+    }
+
+    private void addToEmpty(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        headNode = newNode;
+        tailNode = newNode;
+        size++;
+    }
+
+    private void addFirst(T value) {
+        if (size == 0) {
+            addToEmpty(value);
+        } else {
+            Node<T> newNode = new Node<>(value, null, headNode);
+            headNode.previous = newNode;
+            headNode = newNode;
+            size++;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            addToEmpty(value);
+        } else {
+            Node<T> newNode = new Node<>(value, tailNode, null);
+            tailNode.next = newNode;
+            tailNode = newNode;
+            size++;
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size() || size() == 0) {
```

Comment:
```suggestion
        } else if (index == size()) {
```
if size == 0 index is 0 as well, or it's out of bound